### PR TITLE
minor code cleanup in VGMItem, HexView and DataSeg

### PR DESCRIPTION
**Pull request by mikelow**
Resolves a number of compiler warnings and generally cleans up the code

## Motivation and Context
Trying to drive down all the warnings. Every wee bit helps.

## How Has This Been Tested?
Tested general usage of the app, including HexView.

## Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [x] My code follows the code style of this project.
- [x] I have read the **CONTRIBUTING** document.

### CODE REVIEW

**Comment by mikelow** [comment]
@sykhro I am concerned that I may overwhelm you with PRs to the point this becomes an unrewarding chore for you. I want to give you the chance to review everything, but please don't feel obligated to review them all.

File: src/main/VGMItem.cpp [hunk 1]
```diff
@@ -37,7 +37,7 @@ bool VGMItem::IsItemAtOffset(uint32_t offset, bool includeContainer, bool matchS
   return GetItemFromOffset(offset, includeContainer, matchStartOffset) != nullptr;
 }
 
-VGMItem *VGMItem::GetItemFromOffset(uint32_t offset, bool includeContainer, bool matchStartOffset) {
+VGMItem *VGMItem::GetItemFromOffset(uint32_t offset, bool /*includeContainer*/, bool matchStartOffset) {
```
Comment (sykhro) [inline src/main/VGMItem.cpp:40]: if you want to keep the parameter name, a more readable alternative is using the attribute `[[maybe_unused]]`

File: src/ui/qt/workarea/HexView.cpp [hunk 2]
```diff
@@ -193,7 +193,7 @@ void HexView::changeEvent(QEvent *event) {
     if (!scrollArea) return;
 
     scrollArea->installEventFilter(
-      new LambdaEventFilter([this](QObject* obj, QEvent* event) -> bool {
+      new LambdaEventFilter([this](QObject* /*obj*/, QEvent* event) -> bool {
```
Comment (sykhro) [inline src/ui/qt/workarea/HexView.cpp:196]: ditto

**Comment by sykhro** [comment]
> @sykhro I am concerned that I may overwhelm you with PRs to the point this becomes an unrewarding chore for you. I want to give you the chance to review everything, but please don't feel obligated to review them all.

No worries :)

File: src/main/VGMItem.cpp (hunk 1 above)
Comment (mikelow) [reply src/main/VGMItem.cpp]: 👍

**Review by sykhro — APPROVED** [review]

**Pull request merged**